services/app-api/src/resolvers/user/updateCMSUser.ts

```diff
@@ -54,34 +62,38 @@ export function updateCMSUserResolver(
 
         const stateAssignmentCodes: StateCodeType[] = []
         const invalidCodes = []
+        let invalidStateCodes
 
-        for (const assignment of stateAssignments) {
-            if (isValidStateCode(assignment)) {
-                stateAssignmentCodes.push(assignment)
-            } else {
-                invalidCodes.push(assignment)
+        if (stateAssignments) {
+            for (const assignment of stateAssignments) {
+                if (isValidStateCode(assignment)) {
+                    stateAssignmentCodes.push(assignment)
+                } else {
+                    invalidCodes.push(assignment)
+                }
             }
-        }
 
-        // check that the state codes are valid
-        if (invalidCodes.length > 0) {
-            const invalidStateCodes = stateAssignments.filter(
-                (assignment) => !isValidStateCode(assignment)
-            )
+            // check that the state codes are valid
+            if (invalidCodes.length > 0) {
+                invalidStateCodes = stateAssignments.filter(
+                    (assignment) => !isValidStateCode(assignment)
+                )
 
-            const errMsg = 'Invalid state codes'
-            logError('updateCmsUser', errMsg)
-            setErrorAttributesOnActiveSpan(errMsg, span)
-            throw new UserInputError(errMsg, {
-                argumentName: 'stateAssignments',
-                argumentValues: invalidStateCodes,
-            })
+                const errMsg = 'Invalid state codes'
+                logError('updateCmsUser', errMsg)
+                setErrorAttributesOnActiveSpan(errMsg, span)
+                throw new UserInputError(errMsg, {
+                    argumentName: 'stateAssignments',
+                    argumentValues: invalidStateCodes,
+                })
+            }
         }
-
         const result = await store.updateCmsUserProperties(
             cmsUserID,
             stateAssignmentCodes,
-            divisionAssignment
+            currentUser.id,
+            divisionAssignment,
+            'Updated user assignments' // someday might have a note field and make this a param
         )
```

Comment:
Could you update the way we are throwing errors if `store.updateCmsUserProperties` returns an error? We added a new [error-handling](https://github.com/Enterprise-CMCS/managed-care-review/blob/main/docs/technical-design/error-handling.md) doc in our `technical-desgin` docs.

Take a look at `submitHealthPlanPackage` resolver to see how we handle errors from the DB.

Reply:
Done.  Thanks for the pointer to that doc.